daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/PState.scala

```diff
@@ -220,6 +221,7 @@ final class PState private (
       if (variable.isDefined)
         variable.get.reset
     }}
+    changedVariablesStack.top.clear
```

Comment:
So we're back to having to clear this again now. Please add comment as to why? Not obvious to me that we need this.

Reply:
I have now added a comment to this. The reason why it needs to be cleared is that when parsing a choice (or potentially an unordered sequence) there can be multiple calls to PState.reset() within the same push/popPointOfUncertainty. Which means that the list on changedVariablesStack.top will continue to accumulate changes if the list is not cleared. For example, choice branch 1 reads a variable but ultimately fails calling PState.reset(), which will reset the read on the variable, but then the choice enters branch 2 without popping the POU stack. Branch 2 then sets the variable but ultimately fails too and when PState.reset() gets called it will reset the setVariable call and then try to reset the readVariable call from branch 1 since it hasn't been cleared and we are still in the same POU.